jerry-core/ecma/base/ecma-helpers-conversion.c

```diff
@@ -324,10 +324,78 @@
 
 #define EPSILON 0.0000001
 
+#if CONFIG_ECMA_NUMBER_TYPE == CONFIG_ECMA_NUMBER_FLOAT64
+/**
+ * Number.MAX_VALUE and Number.MIN_VALUE exponent parts while using 64 bit float representation
+ */
+# define NUMBER_MAX_VALUE_EXPONENT 308
+# define NUMBER_MIN_VALUE_EXPONENT -324
+#elif CONFIG_ECMA_NUMBER_TYPE == CONFIG_ECMA_NUMBER_FLOAT32
+/**
+ * Number.MAX_VALUE and Number.MIN_VALUE exponent parts while using 32 bit float representation
+ */
+# define NUMBER_MAX_VALUE_EXPONENT 38
+# define NUMBER_MIN_VALUE_EXPONENT -38
```

Comment:
If we allow subnormals for 64 bit floating point numbers (I guess, that's why the minimum exponent is -324 for doubles, not -308), then we should allow subnormals for 32 bit floating point numbers as well. That would mean -45 as min exponent for CONFIG_ECMA_NUMBER_FLOAT32.

Sources:
* Double precision (https://en.wikipedia.org/wiki/Double-precision_floating-point_format): "The 11 bit width of the exponent allows the representation of numbers between 10^−308 and 10^308, with full 15–17 decimal digits precision. By compromising precision, the subnormal representation allows even smaller values up to about 5 × 10^−324."
* Single precision (https://en.wikipedia.org/wiki/Single-precision_floating-point_format): "The minimum positive normal value is 2^−126 ≈ 1.18 × 10^−38 and the minimum positive (denormal) value is 2^−149 ≈ 1.4 × 10^−45."

And I'd also propose a naming change, to be more in line with IEEE 754 docs (https://en.wikipedia.org/wiki/IEEE_754): what about naming the macros `NUMBER_{MAX,MIN}_DECIMAL_EXPONENT`?

Reply:
That's true, we should use subnormals in both cases. Your renaming proposal might makes the code more esay to understand so I'll change this macro names as well.